src/libraries/System.Memory/src/System/Buffers/SequenceReader.Search.cs

```diff
@@ -578,11 +578,11 @@ public long AdvancePast(T value)
             {
                 // Advance past all matches in the current span
                 int i;
-                for (i = CurrentSpanIndex; i < CurrentSpan.Length && CurrentSpan[i].Equals(value); i++)
+                for (i = _currentSpanIndex; i < CurrentSpanLength && Unsafe.Add(ref CurrentSpanStart, i).Equals(value); i++)
```

Comment:
All of this Unsafe.Add usage makes me extremely nervous.  We're ok using it on really hot paths when it's obvious from local examination that's it's correct, but this isn't local; it'd be really easy for this to accidentally result in buffer overruns / underruns and GC holes.

If the previous version:
```C#
for (i = CurrentSpanIndex; i < CurrentSpan.Length && CurrentSpan[i].Equals(value); i++)
```
were instead rewritten to:
```C#
ReadOnlySpan<T> currentSpan = CurrentSpan;
for (int i = CurrentSpanIndex; (uint)i < (uint)currentSpan.Length && currentSpan[i].Equals(value); i++)
```
the JIT should be able to eliminate the bounds check and it'd be safe.  Or similarly if it were changed to:
```C#
ReadOnlySpan<T> currentSpan = CurrentSpan.Slice(CurrentSpanIndex);
for (int i = 0; i < currentSpan.Length && currentSpan[i].Equals(value); i++);
```
it will also avoid the bounds checks.

I'm obviously all for improving the performance of things, but we need to do it in a maintainable and safe we.  We should strive to avoid all of this unsafe usage.